esm_runscripts/pbs.py

```diff
@@ -0,0 +1,152 @@
+"""
+Contains functions for dealing with PBS-based batch systems
+"""
+import os
+import subprocess
+import sys
+import esm_parser
+
+class Pbs:
+    """
+    Deals with PBS, allowing you to check if a job is submitted, get the
+    current job ID, generate the MOAB commands (e.g. ``aprun``), get the current job
+    state, and check if a job is still running.
+
+
+    Attributes
+    ----------
+    filename : str
+        The filename for srun commands, defaults to ``hostfile_srun``
+    path : str
+        Full path to this file, defaults to ``thisrun_scripts_dir / filename``
+
+    Parameters
+    ----------
+    config : dict
+        The run configuration, needed to determine where the script directory
+        for this particular run is.
+    """
+    def __init__(self, config):
+        # No hostfile for PBS
+        self.filename = ""
+        self.path = ""
+
+    @staticmethod
+    def check_if_submitted():
+        """
+        Determines if a job is submitted in the currently running shell by
+        checking for ``PBS_JOBID`` in the environment
+
+        Returns
+        -------
+        bool
+        """
+        return "PBS_JOBID" in os.environ
+
+    @staticmethod
+    def get_jobid():
+        """
+        Gets the current PBS JOB ID
+
+        Returns
+        -------
+        str or None
+        """
+        return os.environ.get("PBS_JOBID")
```

Comment:
Do we need a default return value here? It would be None anyway,s if it is not submitted, I guess.